drivers/reset/reset_ast10x0.c

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2022 ASPEED Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aspeed_ast10x0_reset
+#include <errno.h>
+#include <zephyr/dt-bindings/reset/ast10x0_reset.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/sys/util.h>
+
+
+/*
+ * RESET_CTRL0/1_ASSERT registers:
+ *   - Each bit in these registers controls a reset line
+ *   - Write '1' to a bit: assert the corresponding reset line
+ *   - Write '0' to a bit: no effect
+ * RESET_CTRL0/1_DEASSERT register:
+ *   - Write '1' to a bit: clear the corresponding bit in RESET_CTRL0/1_ASSERT.
+ *                         (deassert the corresponding reset line)
+ */
+#define RESET_CTRL0_ASSERT		0x40
+#define RESET_CTRL0_DEASSERT		0x44
+#define RESET_CTRL1_ASSERT		0x50
+#define RESET_CTRL1_DEASSERT		0x54
+
+struct reset_aspeed_config {
+	const struct device *syscon;
+};
+
+#define DEV_CFG(dev) ((const struct reset_aspeed_config *const)(dev)->config)
```

Comment:
please remove usage of such macros, this was deleted from all drivers a long time ago.

Reply:
Thank you for your comment.  I removed the macros in the latest PR.